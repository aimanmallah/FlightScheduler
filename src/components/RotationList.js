import { connect } from 'react-redux';

import arrow from '../assets/arrow.png';

const RotationList = (props) => {
  const renderRotations = () => {
    return props.aircraftRotation.map((rotation) => (
      <div key={rotation.id} className="w-80 h-32 rounded overflow-hidden shadow-lg my-2">
        <div className="px-6 py-4 bg-green-100">
          <div className="font-bold text-xl mb-2">Flight: {rotation.id}</div>
          <div className="flex flex-row">
            <div className="flex flex-col">
              <p className="text-grey-darker font-bold text-base">{rotation.origin}</p>
              <p className="text-grey-darker text-base">{rotation.readable_departure}</p>
            </div>
            <div>
              <img src={arrow} className="object-scale-down h-20 w-full" />
            </div>
            <div className="flex flex-col">
              <p className="text-grey-darker font-bold text-base">{rotation.destination}</p>
              <p className="text-grey-darker text-base">{rotation.readable_arrival}</p>
            </div>
          </div>
        </div>
      </div>
    ));
  };

  return (
    <div>
      {props.aircraftRotation.length > 0 && (
        <div className="flex flex-col items-center">{renderRotations()}</div>
      )}
    </div>
  );
};

const mapStateToProps = (state) => {
  return {
    aircraftRotation: state.flightsData.rotation,
  };
};

export default connect(mapStateToProps)(RotationList);
