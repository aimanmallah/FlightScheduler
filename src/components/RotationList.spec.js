import { render } from '../utils/test-utils';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import RotationList from './RotationList';

const rotationData = {
  id: 'AS1001',
  departuretime: 21600,
  arrivaltime: 26100,
  readable_departure: '06:00',
  readable_arrival: '07:15',
  origin: 'LFSB',
  destination: 'LFMN',
};

const mockStore = configureStore([]);

describe('RotationList', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      flightsData: {
        rotation: [rotationData],
      },
    });
  });

  it('renders as expected', () => {
    const { container, getByText } = render(
      <Provider store={store}>
        <RotationList />
      </Provider>
    );

    expect(container.firstChild).toMatchSnapshot();
    expect(getByText(`Flight: ${rotationData.id}`)).toBeTruthy();
    expect(getByText(`${rotationData.readable_departure}`)).toBeTruthy();
    expect(getByText(`${rotationData.readable_arrival}`)).toBeTruthy();
  });
});
