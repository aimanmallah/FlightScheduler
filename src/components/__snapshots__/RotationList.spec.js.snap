// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RotationList renders as expected 1`] = `
<div>
  <div
    class="flex flex-col items-center"
  >
    <div
      class="w-80 h-32 rounded overflow-hidden shadow-lg my-2"
    >
      <div
        class="px-6 py-4 bg-green-100"
      >
        <div
          class="font-bold text-xl mb-2"
        >
          Flight: 
          AS1001
        </div>
        <div
          class="flex flex-row"
        >
          <div
            class="flex flex-col"
          >
            <p
              class="text-grey-darker font-bold text-base"
            >
              LFSB
            </p>
            <p
              class="text-grey-darker text-base"
            >
              06:00
            </p>
          </div>
          <div>
            <img
              class="object-scale-down h-20 w-full"
              src="arrow.png"
            />
          </div>
          <div
            class="flex flex-col"
          >
            <p
              class="text-grey-darker font-bold text-base"
            >
              LFMN
            </p>
            <p
              class="text-grey-darker text-base"
            >
              07:15
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
